### PR TITLE
Support the yield clause when fetch vertices on *.

### PR DESCRIPTION
**Pull request by Shylock-Hg**
A part of https://github.com/vesoft-inc/nebula/pull/2222

### CODE REVIEW

File: src/validator/FetchVerticesValidator.h [hunk 1]
```diff
@@ -33,21 +33,17 @@ class FetchVerticesValidator final : public Validator {
...
 private:
-    GraphSpaceID spaceId_{0};
     DataSet srcVids_{{kVid}};  // src from constant
     Expression* srcRef_{nullptr};  // src from runtime
     Expression* src_{nullptr};  // src in total
-    std::string tagName_;
-    // none if not specified tag
-    folly::Optional<TagID> tagId_;
-    std::shared_ptr<const meta::SchemaProviderIf> schema_;
+    bool onStar_{false};
```
Comment (yixinglu) [inline src/validator/FetchVerticesValidator.h:44]: suggest a meaningful name related to all tags

**Review by yixinglu — COMMENTED** [review]
others LGTM

File: src/validator/FetchVerticesValidator.h (hunk 1 above)
Comment (Shylock-Hg) [reply src/validator/FetchVerticesValidator.h:44]: There are two `allTags` in here, one extend from * , the other means user type all tags.

File: src/validator/FetchVerticesValidator.cpp [hunk 2]
```diff
@@ -186,67 +215,27 @@ Status FetchVerticesValidator::preparePropertiesWithYield(const YieldClause *yie
...
-        colNames_.emplace_back(VertexID);
-        gvColNames_.emplace_back(VertexID);   // keep compatible with 1.0
-        for (std::size_t i = 0; i < schema_->getNumFields(); ++i) {
-            outputs_.emplace_back(schema_->getFieldName(i),
-                                  SchemaUtil::propTypeToValueType(schema_->getFieldType(i)));
-            colNames_.emplace_back(tagName_ + '.' + schema_->getFieldName(i));
+    outputs_.emplace_back(VertexID, Value::Type::STRING);
+    colNames_.emplace_back(VertexID);
+    gvColNames_.emplace_back(colNames_.back());
+    for (const auto &tagSchema : tagsSchema_) {
```
Comment (laura-ding) [inline src/validator/FetchVerticesValidator.cpp:221]: you can use `tags_ ` directly

File: src/validator/FetchVerticesValidator.cpp [hunk 3]
```diff
@@ -186,67 +215,27 @@ Status FetchVerticesValidator::preparePropertiesWithYield(const YieldClause *yie
...
+        storage::cpp2::VertexProp vProp;
+        vProp.set_tag(tagSchema.first);
+        auto tagNameResult = qctx_->schemaMng()->toTagName(space_.id, tagSchema.first);
```
Comment (laura-ding) [inline src/validator/FetchVerticesValidator.cpp:224]: and here is no need `toTagName `.

File: src/validator/test/FetchVerticesTest.cpp [hunk 4]
```diff
@@ -241,6 +241,80 @@ TEST_F(FetchVerticesValidatorTest, FetchVerticesProp) {
         auto result = Eq(qctx->plan()->root(), gv);
         ASSERT_TRUE(result.ok()) << result;
     }
+    // ON * with yield
+    {
+        auto qctx = getQCtx("FETCH PROP ON * \"1\", \"2\" YIELD person.name");
```
Comment (laura-ding) [inline src/validator/test/FetchVerticesTest.cpp:246]: I think the syntax here is contradictory. we can make it more meaningful.
such as
```
FETCH PROP ON person  \"1\", \"2\" YIELD name
```

File: src/validator/FetchVerticesValidator.cpp (hunk 2 above)
Comment (Shylock-Hg) [reply src/validator/FetchVerticesValidator.cpp:221]: Can't I need the ordered map.

File: src/validator/test/FetchVerticesTest.cpp (hunk 4 above)
Comment (Shylock-Hg) [reply src/validator/test/FetchVerticesTest.cpp:246]: This PR keep same with 1.0. We could consider it in later.

**Review by yixinglu — APPROVED** [review]

File: src/parser/TraverseSentences.h [hunk 5]
```diff
@@ -307,16 +307,18 @@ class FetchVerticesSentence final : public Sentence {
         yieldClause_.reset(clause);
     }
 
-    explicit FetchVerticesSentence(Expression *ref) {
+    explicit FetchVerticesSentence(Expression *ref, YieldClause  *clause) {
```
Comment (yixinglu) [inline src/parser/TraverseSentences.h:310]: remove explicit

File: src/parser/TraverseSentences.h [hunk 6]
```diff
@@ -307,16 +307,18 @@ class FetchVerticesSentence final : public Sentence {
...
         kind_ = Kind::kFetchVertices;
         tag_ = std::make_unique<std::string>("*");
         vidRef_.reset(ref);
+        yieldClause_.reset(clause);
     }
 
-    explicit FetchVerticesSentence(VertexIDList *vidList) {
+    explicit FetchVerticesSentence(VertexIDList *vidList, YieldClause  *clause) {
```
Comment (yixinglu) [inline src/parser/TraverseSentences.h:317]: ditto

File: src/parser/TraverseSentences.h (hunk 5 above)
Comment (Shylock-Hg) [reply src/parser/TraverseSentences.h:310]: This will disable the implicit convert ctor.

**Review by laura-ding — APPROVED** [review]

**Pull request merged**